### PR TITLE
docs: add homepage link

### PR DESCRIPTION
**Pull request by DanArthurGallagher**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | No issue raised that I can see.
| Patch: Bug Fix?          | I haven't incremented the version because there is no functional change.
| Major: Breaking Change?  | No.
| Minor: New Feature?      | No.
| Tests Added + Pass?      | There are no tests in the package. I ran the global tests and none of the reported errors were in this package.
| Documentation PR Link    | This is the documentation change.
| Any Dependency Changes?  | No.
| License                  | MIT

Adds a homepage link to populate npmjs.com from the `package.json` so that a link is available there in the sidebar. This link is exactly the one taken from `README.md`.

Update the repository property syntax to the object syntax including `type: "git"`.

### CODE REVIEW

File: packages/babel-runtime/package.json [hunk 1]
```diff
@@ -6,7 +6,11 @@
   "publishConfig": {
     "access": "public"
   },
-  "repository": "https://github.com/babel/babel/tree/master/packages/babel-runtime",
+  "repository": {
```
Comment (JLHwung) [inline packages/babel-runtime/package.json]: On the `repository`, NPM offers an extra [`directory` field](https://docs.npmjs.com/files/package.json#repository) for monorepo:

```json
"repository": {
  "type" : "git",
  "url" : "https://github.com/babel/babel.git",
  "directory": "packages/babel-runtime"
}
```
To be consistent across different packages in this repo, I think we had better to replace all the repository fields.

**Comment by JLHwung** [comment]
`npm docs` will respect the `homepage` field, I think it is a good idea to align `homepage` to the README.

File: packages/babel-runtime/package.json (hunk 1 above)
Comment (DanArthurGallagher) [reply packages/babel-runtime/package.json]: Thanks for pointing that out! I will make the update here.

**Comment by DanArthurGallagher** [comment]
Hi there @JLHwung, I'm not sure of what you meant by [your review comment](https://github.com/babel/babel/pull/10418#issuecomment-529508652) 

> &hellip; I think it is a good idea to align `homepage` to the README in all the sub packages.

Do you mean that I should link directly to the README in the `homepage` link, or that the `homepage` of the other submodules should also point directly to their documentation pages?

I can make either change, but I did think that as a consumer it would be useful to have a link directly to the intended location rather than replicate the link from the README.
However the npmjs.com sidebar UI would be populated as long as something is added, and that was my intention.

File: packages/babel-runtime/package.json [hunk 2]
```diff
@@ -6,7 +6,12 @@
...
+    "type" : "git",
+    "url" : "https://github.com/babel/babel.git",
+    "directory": "packages/babel-runtime"
+  },
+  "homepage": "https://babeljs.io/docs/en/next/babel-runtime.html",
```
Comment (JLHwung) [inline packages/babel-runtime/package.json]: nit: `.html` is not necessary.

**Comment by nicolo-ribaudo** [comment]
@DanArthurGallagher It would be really useful is the "homepage" link pointing to the correct page on the website is added to all the packages which have a docs page!

**Pull request merged**